ui/admin-portal/src/app/components/candidates/view/tasks/view-candidate-tasks.component.ts

```diff
@@ -102,6 +105,31 @@ export class ViewCandidateTasksComponent implements OnInit, OnChanges {
 
   }
 
+  async assignDuolingoCouponTask() {
+    const duolingoTask = await this.taskService.listTasks().toPromise();
```

Comment:
This is not a pattern that we use. We don't use async because we don't need to when using Observables and subscribe. Look at other examples in our code where we are calling an API.

Similarly we very rarely use Promises - typically only when we are using an old library that requires us to use Promises.

Reply:
Looking at this again @MuhannedNoman - I can see your thumbs up - but no changes that I can approve.